packages/frontend/src/features/comments/components/CommentWithMentions/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { useMantineTheme } from '@mantine/core';
+import { RichTextEditor } from '@mantine/tiptap';
+import Mention from '@tiptap/extension-mention';
+import Placeholder from '@tiptap/extension-placeholder';
+import { JSONContent, useEditor } from '@tiptap/react';
+import StarterKit from '@tiptap/starter-kit';
+import { FC, useEffect } from 'react';
+import { SuggestionsItem } from '../../types';
+import { generateSuggestionWrapper } from './generateSuggestionWrapper';
+
+type Props = {
+    readonly: boolean;
+    suggestions?: SuggestionsItem[];
+    setCommentText?: (text: string) => void;
+    setCommentHtml?: (html: string) => void;
+    content?: string;
+    shouldClearEditor?: boolean;
+    setShouldClearEditor?: (shouldClearEditor: boolean) => void;
+    setMentions?: (mentions: string[]) => void;
+};
+
+const parseMentions = (data: JSONContent): string[] => {
+    const mentions = (data.content || []).flatMap(parseMentions);
+    if (data.type === 'mention' && data.attrs?.id) {
+        mentions.push(data.attrs.id);
+    }
+
+    const uniqueMentions = [...new Set(mentions)];
+
+    return uniqueMentions;
+};
+
+export const CommentWithMentions: FC<Props> = ({
+    readonly,
+    suggestions,
+    setMentions,
+    setCommentText,
+    setCommentHtml,
+    content,
+    shouldClearEditor,
+    setShouldClearEditor,
+}) => {
+    const theme = useMantineTheme();
+
+    const editor = useEditor({
+        editable: !readonly,
+        extensions: [
+            StarterKit,
+            Mention.configure({
+                HTMLAttributes: {
+                    style: `color: ${theme.colors.blue['6']}; font-weight: 500;`,
+                },
+                suggestion: suggestions
+                    ? generateSuggestionWrapper(suggestions)
+                    : undefined,
+            }),
+            Placeholder.configure({
+                placeholder: 'Add comment (type @ to mention someone)',
+            }),
+        ],
+        content,
+        onUpdate(e) {
+            setCommentText?.(e.editor.getText());
+            setCommentHtml?.(e.editor.getHTML());
+            setMentions?.(parseMentions(e.editor.getJSON()));
+        },
```

Comment:
these should not be called on `onUpdate` but only when the user hits submit

Reply:
yeah, that makes sense, is there any problem on doing that ?

Reply:
I made a commit for this, feel free to revert if you don't like it https://github.com/lightdash/lightdash/pull/9103/commits/174572cb8e3d2f379190c46031f7f49a15cf6aa9

Reply:
nice, that's great, thanks for doing it 🙏